packages/core/src/app/customer/CheckoutButtonV2.spec.tsx

```diff
@@ -0,0 +1,71 @@
+import { CheckoutButtonProps } from '@bigcommerce/checkout-js/payment-integration';
+import { createCheckoutService, createLanguageService } from '@bigcommerce/checkout-sdk';
+import { mount } from 'enzyme';
+import React from 'react';
+
+import CheckoutButton from './CheckoutButtonV2';
+
+describe('CheckoutButton', () => {
+    let defaultProps: CheckoutButtonProps;
+
+    beforeEach(() => {
+        const checkoutService = createCheckoutService();
+
+        defaultProps = {
+            checkoutService,
+            checkoutState: checkoutService.getState(),
+            containerId: 'button-container',
+            language: createLanguageService(),
+            methodId: 'foobar',
+            onUnhandledError: jest.fn(),
+        };
+    });
+
+    it('initializes when component is mounted', () => {
+        const { checkoutService, onUnhandledError } = defaultProps;
+
+        jest.spyOn(checkoutService, 'initializeCustomer')
+            .mockResolvedValue(checkoutService.getState());
+
+        mount(<CheckoutButton { ...defaultProps } />);
+
+        expect(checkoutService.initializeCustomer)
+            .toHaveBeenCalledWith({
+                methodId: 'foobar',
+                foobar: {
+                    container: 'button-container',
+                    onUnhandledError,
+                },
+            });
+    });
+
+    it('deinitializes when component is unmounted', () => {
```

Comment:
Great work. The resolving mechanism is now better organized.

Just a question about wallet button itself: when should we expect a wallet button to be deinitialized or unmounted?

I can see Bolt deinitializes the previous Bolt customer strategy each time before initialization.

https://github.com/bigcommerce/checkout-js/blob/42ee27d679f81eaabc8b43771b08575525962d4a/packages/core/src/app/customer/checkoutSuggestion/BoltCheckoutSuggestion.tsx#L30

Reply:
`deinitialize` should only need to be called for components that have been initialized. In other words, when a component gets unmounted. We'll need to revisit Bolt once when get to creating an integration point for checkout suggestions. The scope of this PR is limited to just wallet buttons (which Bolt doesn't have).

Reply:
@bc-peng Bolt doesn't work as a wallet button at the moment, it is encapsulated in it's own component called `<CheckoutSuggestion />` which is snowflake and should be fixed going forward separately.